ttnn/ttnn/operations/core.py

```diff
@@ -734,14 +734,23 @@ def requires_padding_change(layout, shape):
                     output_mem_config = ttl.tensor.MemoryConfig(
                         memory_layout_config.memory_layout, ttl.tensor.BufferType.L1
                     )
+                    if memory_config is not None:
+                        output_mem_config = memory_config
                     output_tensor = ttl.tensor.untilize_with_unpadding(
                         input_tensor,
                         (0, 0, 0, 0),
                         intended_4D_shape,
                         output_mem_config,
                     )
                 else:
-                    output_tensor = ttl.tensor.untilize_with_unpadding(input_tensor, (0, 0, 0, 0), intended_4D_shape)
+                    if memory_config is None:
```

Comment:
Do you need the if-else statement? Can you not just pass in the memory config?

Reply:
I mean doing this in both cases:
```
output_tensor = ttl.tensor.untilize_with_unpadding(
       input_tensor, (0, 0, 0, 0), intended_4D_shape, memory_config
 )
```

If the C++ doesn't take None, then please update it instead of doing this